### PR TITLE
React native filter local

### PR DESCRIPTION
**Pull request by dennisdeng2002**

### CODE REVIEW

File: react-native/app/action-creators/restaurants.js [hunk 1]
```diff
@@ -65,9 +64,10 @@ export const getRestaurants = () =>
 				})
 				.then(res => res.data)
 				.then(body => {
-					if(!body.businesses.length) dispatch(setAvailable(false));
-					else dispatch(receiveRestaurants(body.businesses.sort(() => 0.5 - Math.random())));
-					resolve();
+					let restaurants = body.businesses;
+					if(!restaurants.length) dispatch(setAvailable(false));
+					else dispatch(receiveRestaurants(restaurants.sort(() => 0.5 - Math.random())));
+					resolve(restaurants);
```
Comment (mhario) [inline react-native/app/action-creators/restaurants.js]: I would love to see some comments through here. It looks like the sort call would swap a few cards around randomly but its not really clear

File: react-native/app/components/AdditionalCategories/AdditionalCategories.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+class AdditionalCategories extends Component {
+  constructor(props){
+    super(props);
+    this._renderRow = this._renderRow.bind(this);
+    this.selectOrDeselectCategory = this.selectOrDeselectCategory.bind(this);
+
+    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
+
+    const chosenCategories = props.temporaryCategories || props.chosenCategories;
```
Comment (mhario) [inline react-native/app/components/AdditionalCategories/AdditionalCategories.js]: looking at this line, I am thinking it would take temporaryCategories every time and short circuit the other part. It might be fine but I can't tell -- add a comment here too please :)

File: react-native/app/components/AdditionalCategories/AdditionalCategories.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+     const dataSource = this.state.ds.cloneWithRows(this.state.categories.slice());
+     this.setState({categorySwitchStates, dataSource});
+    };
+  }
+
+  componentWillUnmount(){
+    const categorySwitchStates = this.state.categorySwitchStates;
+    const categories = [];
+
+    Object.keys(categorySwitchStates).forEach((category) => {
```
Comment (mhario) [inline react-native/app/components/AdditionalCategories/AdditionalCategories.js]: Here's another spot where a comment would help. I'm sure this works but I'm not really sure what its doing

File: react-native/app/components/AutocompleteExample.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+              
+            </Text>
+          )}
+        </View>
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
```
Comment (mhario) [inline react-native/app/components/AutocompleteExample.js]: I'm not sure if this file is being used, but if it is it's name should be changed to something more descriptive, and these styles should probably be moved into their own file, just because this one is pretty big. (Or is this file not being used? Because then I'd like to see it deleted altogether)

File: react-native/app/components/Categories/Categories.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    let categorySwitchStates = this.state.categorySwitchStates;
+
+    categorySwitchStates[rowData] = value;
+
+    const dataSource = this.state.ds.cloneWithRows(this.state.chosenCategories.slice());
+
+    this.setState({categorySwitchStates, dataSource});
+  };
+
+  _renderRow(rowData){
```
Comment (mhario) [inline react-native/app/components/Categories/Categories.js]: Some of the stuff in this file looks like it is duplicate with the content of AdditionalCategories. It could be a later refactor, but I wonder if some of that duplicate code could be made into its own component

File: react-native/app/components/Favorites/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { connect } from 'react-redux';
+import Favorites from './Favorites';
+
+import {getFavorites} from '../../action-creators/restaurants';
+// import {getCurrentLocation, getSearchSettings, addSearchSettings, receiveSearchSettings} from '../../action-creators/filter';
+
+const mapStateToProps = state => {
+  return {
+    // location: state.filter.location,
```
Comment (mhario) [inline react-native/app/components/Favorites/index.js]: A lot of dead code here

File: react-native/app/components/Filter/Filter.js [hunk 7]
```diff
@@ -6,49 +6,181 @@ import {
...
+    	});
+    	if(!restaurants.length) Alert.alert('', 'No restaurants found');
+    	else {
+    		this.props.setAvailable(true);
+    		Actions.pop();
+    	};
+    })
+    .catch(console.log);
+	}
+
```
Comment (mhario) [inline react-native/app/components/Filter/Filter.js]: I'm curious why you chose to convert meters to miles this way? Again, I'm not saying its wrong, just wondering why you didn't just multiply by the conversion factor? I also keep changing my own mind about using the 'utils' file, if we were to use the same function.

File: react-native/app/components/Filter/Filter.js [hunk 8]
```diff
@@ -6,49 +6,181 @@ import {
...
-          		  value={5}
+		      		maximumValue={25}
+		      		value={this.state.radius}
 		          {...this.state}
 		          onSlidingComplete={(value) => this.setState({ radius: value })} />
-			</View>
-		);
-	};
+			    </View>
+			    <View>
```
Comment (mhario) [inline react-native/app/components/Filter/Filter.js]: I think you might have a little tab-formatting issue here

File: react-native/app/components/Login/Login.android.js [hunk 9]
```diff
@@ -140,7 +141,8 @@ export default class Login extends Component {
 
           <Button title="Login" onPress={login} buttonStyle={styles.login}/>
           
-        </KeyboardAvoidingView>
+        </View>
+        <KeyboardSpacer onToggle={(a,b)=>console.log(a,b)}/>
```
Comment (mhario) [inline react-native/app/components/Login/Login.android.js]: why are we console logging here? I'm guessing you were working on getting the keyboard to not obscure the input boxes which is cool but this looks unfinished

File: react-native/app/components/Restaurant/styles.js [hunk 10]
```diff
@@ -12,31 +12,36 @@ export const styles = StyleSheet.create({
...
   },
   reviewImage: {
     width: 30,
     height: 30,
     borderRadius: 15
   },
   tab: {
-    flex: 1,
-    height: 850,
+    width: Dimensions.get('window').width * 1,
```
Comment (mhario) [inline react-native/app/components/Restaurant/styles.js]: curious what the *1s are for lol

File: react-native/app/components/SwipeView/HighlightsBox.js [hunk 11]
```diff
@@ -28,17 +28,16 @@ export const HighlightsBox = props => {
   formatPrice = () => {
     switch (props.restaurant.price) {
       case '$': 
-        return (<View style={styles.priceBox}><Text style={styles.priceBold}>$<Text style={styles.priceLight}>$$$</Text></Text></View>);
+        return (<View style={styles.priceBox}><Text style={styles.priceLight}>$<Text style={styles.priceBold}>$$$</Text></Text></View>);
```
Comment (mhario) [inline react-native/app/components/SwipeView/HighlightsBox.js]: why did you change this? it is backwards now

File: react-native/app/components/SwipeView/SwipeView.js [hunk 12]
```diff
@@ -3,22 +3,28 @@ import { SummaryCard } from './SummaryCard';
...
   constructor(props) {
     super(props);
-    this.state = {
-      startIndex: props.swipeCounter
-    };
     this.onSwipeRight = this.onSwipeRight.bind(this);
     this.onSwipeLeft = this.onSwipeLeft.bind(this);
   }
 
+  _compareArrays(arr1, arr2) {
```
Comment (mhario) [inline react-native/app/components/SwipeView/SwipeView.js]: I wonder if we can do this without iterating across every entry every time? This is fine, I just wonder if it could be the target of a refactor later--especially if we see the swipeview scene starts taking longer to load

File: react-native/ios/biteswipe.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -1210,6 +1216,7 @@
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = YES;
+				DEVELOPMENT_TEAM = 69P7UKJ6LW;
```
Comment (mhario) [inline react-native/ios/biteswipe.xcodeproj/project.pbxproj]: I'm curious what this is?

File: react-native/app/components/AdditionalCategories/AdditionalCategories.js [hunk 14]
```diff
@@ -0,0 +1,118 @@
+import Autocomplete from 'react-native-autocomplete-input';
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/AdditionalCategories/AdditionalCategories.js]: This file could use more commenting.

File: react-native/app/components/AutocompleteExample.js [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      films: [],
+      query: ''
+    };
+  }
+
+  componentDidMount() {
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/AutocompleteExample.js]: do we need this file? lot of stuff about films in here.

File: react-native/app/components/Restaurant/CarouselItem.js [hunk 16]
```diff
@@ -1,19 +1,16 @@
...
   return(
     <View style={styles.container}>
       <Image
         source={{uri: props.image}}
         style={styles.cardImage}>
-        <LinearGradient
-          colors={['transparent', '#777']}
-          start={{x:0, y:.8}}
-          style={styles.cardImage} />
+        <View style={styles.cardTitleContainer}>
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/Restaurant/CarouselItem.js]: Wondering why we took off the linear gradient.

File: react-native/app/components/SwipeView/SwipeView.js [hunk 17]
```diff
@@ -36,15 +42,40 @@ export default class SwipeView extends Component {
...
-    if(!slicedRestaurants.length || this.props.swipeCounter === this.props.restaurants.length) {
+    if(!slicedRestaurants.length || this.props.swipeCounter >= this.props.restaurants.length) {
       return (
         <View style={styles.swipeViewBackground}>
-          <NoMoreCards getRestaurants={this.props.getRestaurants}/>
+          <NoMoreCards/>
         </View>
         );
+    } else if(slicedRestaurants.length === 1) {
+      // fucking hax - DeckSwiper requires two 'cards' so we push a empty card into array
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/SwipeView/SwipeView.js]: gotta change this comment

File: react-native/app/components/Categories/Categories.js (hunk 5 above)
Comment (dennisdeng2002) [inline react-native/app/components/Categories/Categories.js]: yeah I agree

File: react-native/app/components/SwipeView/HighlightsBox.js (hunk 11 above)
Comment (dennisdeng2002) [inline react-native/app/components/SwipeView/HighlightsBox.js]: I think it's easier to see since the highlight box is light green.

Comment (mhario) [inline react-native/app/components/SwipeView/HighlightsBox.js]: I was about to change the color this afternoon, that was kind of a hacky temporary color. Either way, I think it'd be better to change that in the styles rather than using 'bold' on non-data

File: react-native/app/components/DrawerLayout/DrawerLayout.js [hunk 18]
```diff
@@ -53,7 +53,10 @@ export default class DrawerLayout extends Component {
 	  		initialized: true
 	  	});
 	  });
+	}
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/DrawerLayout/DrawerLayout.js]: is this necessary?

File: react-native/app/components/DrawerLayout/DrawerLayout.js [hunk 19]
```diff
@@ -53,7 +53,10 @@ export default class DrawerLayout extends Component {
...
 
+	componentWillUnmount() {
+		console.log('unmounting');
 	}
```
Comment (rachel-rocks-at-ruby) [inline react-native/app/components/DrawerLayout/DrawerLayout.js]: Is this necessary?
